src/Features/Core/Portable/MoveDeclarationNearReference/AbstractMoveDeclarationNearReferenceService.cs

```diff
@@ -61,7 +69,7 @@ private async Task<State> ComputeStateAsync(Document document, SyntaxNode node,
         public async Task<Document> MoveDeclarationNearReferenceAsync(
             Document document, SyntaxNode localDeclarationStatement, CancellationToken cancellationToken)
         {
-            var state = await ComputeStateAsync(document, localDeclarationStatement, cancellationToken);
+            var state = await ComputeStateAsync(document, localDeclarationStatement, canMovePastOtherDeclarationStatements: true, cancellationToken);
```

Comment:
is this not breaking existing behavior?

Reply:
It is expected that the caller first invoke `CanMoveXXXAsynx()` with the required value of `canMovePastOtherDeclarationStatements` if they want to skip moving the declaration past other declaration statements. The actual request to move the declaration will then move the declaration, and doesn't need same checks again.

---
In reply to: [230603957](https://github.com/dotnet/roslyn/pull/30777#discussion_r230603957) [](ancestors = 230603957)